src/components/shared/VirtualList.js

```diff
@@ -141,59 +159,83 @@ class VirtualListInner extends PureComponent {
   }
 }
 
-VirtualListInner.propTypes = {
-  itemHeight: PropTypes.number.isRequired,
-  className: PropTypes.string,
-  renderItem: PropTypes.func.isRequired,
-  items: PropTypes.array.isRequired,
-  specialItems: PropTypes.array.isRequired,
-  visibleRangeStart: PropTypes.number.isRequired,
-  visibleRangeEnd: PropTypes.number.isRequired,
-  columnIndex: PropTypes.number.isRequired,
+type VirtualListProps = {|
+  +itemHeight: CssPixels,
+  +className: string,
+  +renderItem: RenderItem,
+  +items: *[],
+  +focusable: boolean,
+  +specialItems: *[],
+  +onKeyDown: KeyboardEvent => void,
+  +onCopy: Event => void,
+  +disableOverscan: boolean,
+  +columnCount: number,
+|};
+
+type Geometry = {
+  // getBoundingClientRect in the Flow definitions is wrong, and labels the return values
+  // as a ClientRect, and not a DOMRect. https://github.com/facebook/flow/issues/5475
+  //
+  // Account for that here:
+  outerRect: DOMRect | ClientRect,
```

Comment:
I've seen in other places a DOMRectInterface too (from `utils/dom-rect`); but I believe a DOMRect is good here actually.

I believe the comment is not completely accurate: the Flow definition is not completely wrong, as getBoundingClientRect returns either ClientRect (in Internet Explorer) or DOMRect (in compliant browsers). I think Flow should use `ClientRect  | DOMRect` itself to account for this, though...

That said, because we don't use x and y, maybe we can just type this as a ClientRect...

Reply:
The problem is that Flow applies nominal typing for classes, and not structural, so DOMRect is not equivalent to ClientRect even if the structure is fine, that's why there is the awkward `DOMRectInterface`, as that applies structural typing.